packages/ozone/src/api/admin/searchRepos.ts

```diff
@@ -0,0 +1,42 @@
+import { Server } from '../../lexicon'
+import AppContext from '../../context'
+import { mapDefined } from '@atproto/common'
+
+export default function (server: Server, ctx: AppContext) {
+  server.com.atproto.admin.searchRepos({
+    auth: ctx.roleVerifier,
+    handler: async ({ params }) => {
+      const modService = ctx.modService(ctx.db)
+
+      // prefer new 'q' query param over deprecated 'term'
+      const query = params.q ?? params.term
+
+      // special case for did searches - do exact match
+      if (query?.startsWith('did:')) {
+        const repos = await modService.views.repos([query])
+        const found = repos.get(query)
+        return {
+          encoding: 'application/json',
+          body: {
+            repos: found ? [found] : [],
+          },
+        }
+      }
+
+      const res = await ctx.appviewAgent.api.app.bsky.actor.searchActors(params)
```

Comment:
since we don't maintain an index of all actors in ozone, i just fell back to the search actors route on the appview (but preserved the special case for dids)